### PR TITLE
🚨🚨 Source AirTable: parse formula to simple data types

### LINKED ISSUE #26528: Source Airtable: fix the data types in formula, rollup, lookup fields
### What area the feature impact?

Connectors

## Tell us about the problem you're trying to solve
For our Airtable Source, Integration Data Type: `Formula`, `Rollup` and `Lookup` is displayed as an `array<any>` type.  
By the [Airtable API](https://airtable.com/developers/web/api/field-model#formula) can return a type of the result.

## Describe the solution you’d like
Change the type to number or string depending on the result type for the  `Formula`, `Rollup` and `Lookup` fields.

### PR DESCRIPTION
**Pull request by artem1205**
## What

Resolve https://github.com/airbytehq/airbyte/issues/26528

## How

parse formula to simple data types : `string` or `number`

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨

schema changes: formulas with simple output will be presented as `string` or `number` respectively.

## Pre-merge Actions

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

**Comment by airbyte-oss-build-runner** [comment]
##  source-airtable test report (commit [`7cb0c97b13`](https://github.com/airbytehq/airbyte/pull/31181/commits/7cb0c97b1325b4ea1e52c33187fafe193da0cf56)) - ❌

⏲️  Total pipeline duration: 02mn36s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Connector package install                                       | ✅        |
| Build source-airtable docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Acceptance tests                                                | ❌        |
| Code format checks                                              | ✅        |
| Validate metadata for source-airtable                           | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/artem1205_source-airtable-handle-formula-correctly/1696866214/7cb0c97b1325b4ea1e52c33187fafe193da0cf56/source-airtable/4.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:7cb0c97b1325b4ea1e52c33187fafe193da0cf56)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-airtable test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-airtable test report (commit [`e45d197f52`](https://github.com/airbytehq/airbyte/pull/31181/commits/e45d197f52eb7807237d49eb11363eaed99b822c)) - ✅

⏲️  Total pipeline duration: 01mn38s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Connector package install                                       | ✅        |
| Build source-airtable docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Acceptance tests                                                | ✅        |
| Code format checks                                              | ✅        |
| Validate metadata for source-airtable                           | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/artem1205_source-airtable-handle-formula-correctly/1696868547/e45d197f52eb7807237d49eb11363eaed99b822c/source-airtable/4.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:e45d197f52eb7807237d49eb11363eaed99b822c)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-airtable test
```

File: docs/integrations/sources/airtable.md [hunk 1]
```diff
@@ -111,7 +111,8 @@ See information about rate limits [here](https://airtable.com/developers/web/api
 
 | Version | Date       | Pull Request                                             | Subject                                                         |
 |:--------|:-----------|:---------------------------------------------------------|:----------------------------------------------------------------|
-| 3.0.1   | 2023-05-10 | [25946](https://github.com/airbytehq/airbyte/pull/25946) | Skip stream if it does not appear in catalog                                  |
+| 4.0.0   | 2023-10-09 | [31181](https://github.com/airbytehq/airbyte/pull/31181) | Parse formula to simple data types                              |
```
Comment (bazarnov) [inline docs/integrations/sources/airtable.md:114]: nit: added additional schema processing for the `FORMULA` schema type.

**Review by bazarnov — APPROVED** [review]

**Comment by airbyte-oss-build-runner** [comment]
##  source-airtable test report (commit [`e45e36e35b`](https://github.com/airbytehq/airbyte/pull/31181/commits/e45e36e35b7af1c9679b3d6b43f69e54e3d40774)) - ❌

⏲️  Total pipeline duration: 01mn32s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Connector package install                                       | ✅        |
| Build source-airtable docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Acceptance tests                                                | ❌        |
| Code format checks                                              | ✅        |
| Validate metadata for source-airtable                           | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/artem1205_source-airtable-handle-formula-correctly/1696886594/e45e36e35b7af1c9679b3d6b43f69e54e3d40774/source-airtable/4.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:e45e36e35b7af1c9679b3d6b43f69e54e3d40774)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-airtable test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-airtable test report (commit [`31eac02229`](https://github.com/airbytehq/airbyte/pull/31181/commits/31eac022292b282d963fcd22d6ddf54f48e05c13)) - ✅

⏲️  Total pipeline duration: 02mn18s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Connector package install                                       | ✅        |
| Build source-airtable docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Acceptance tests                                                | ✅        |
| Code format checks                                              | ✅        |
| Validate metadata for source-airtable                           | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/artem1205_source-airtable-handle-formula-correctly/1696887068/31eac022292b282d963fcd22d6ddf54f48e05c13/source-airtable/4.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:31eac022292b282d963fcd22d6ddf54f48e05c13)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-airtable test
```

**Pull request merged**